### PR TITLE
Fix `.wasm` product paths for `wasm32-unknown-none` triple

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Absence of this extension on produced binaries makes it inconvenient to work with triple, as you have to add that extension manually after the fact. For examples, HTTP servers frequently don't infer correct mime-type for these binaries, which confuses browsers that try to load these binaries.

### CODE REVIEW

File: Tests/BuildTests/BuildPlanTests.swift [hunk 1]
```diff
@@ -3431,111 +3431,6 @@ final class BuildPlanTests: XCTestCase {
         XCTAssertMatch(executablePathExtension, "exe")
     }
 
-    func testWASITarget() throws {
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:3434]: Moved to `Tests/BuildTests/CrossCompilationBuildPlanTests.swift` to make `Tests/BuildTests/BuildPlanTests.swift` smaller and increase parallelization during testing, as separate `XCTestCase` types can be parallelized, while multiple functions in `XCTestCase` can't.

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: Sources/Basics/Triple+Basics.swift [hunk 2]
```diff
@@ -148,6 +148,10 @@ extension Triple {
     }
 
     public var executableExtension: String {
+        guard ![.wasm32, .wasm64].contains(self.arch) else {
```
Comment (kateinoigakukun) [inline Sources/Basics/Triple+Basics.swift:151]: How about adding `isWasm` like llvm's one?: https://github.com/llvm/llvm-project/blob/031f9f331723e6bebc405ffdee4b8a87a5fc0472/llvm/include/llvm/TargetParser/Triple.h#L1005

I think `[.wasm32, .wasm64].contains(self.triple.arch)` appears repeatedly, so it's worth adding memory space size independent predicate method.

Comment (MaxDesiatov) [reply Sources/Basics/Triple+Basics.swift]: Good point, I've added it on an extension in `Basics/Triple+Basics.swift`

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by kateinoigakukun — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**